presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPageSourceProvider.java

```diff
@@ -88,6 +104,31 @@ public ConnectorPageSource createPageSource(
         PinotSplit pinotSplit = (PinotSplit) split;
         checkArgument(pinotSplit.getConnectorId().equals(connectorId), "split is not for this connector");
 
+        PinotQueryGenerator.GeneratedPinotQuery brokerQuery = pinotSplit.getBrokerPinotQuery().get();
+
+        if (splitContext.getDynamicFilterPredicate().isPresent()) {
+            brokerQuery = processDynamicFilterSqlQuery(splitContext, brokerQuery, PinotSessionProperties.getMaxPushdownDynamicFilterSize(session));
+        }
+
+        // either no dynamic filter or dynamic filter is too large
+        if (brokerQuery.getQuery().contains(DYNAMIC_FILTER_FUNCTION_TEMPLATE)) {
```

Comment:
What would happen if the query text already contains the text `__DYNAMIC_FILTER_FUNCTION__` as a constant ? Do we escape/guard against this ?

Reply:
We don't do that currently, but also something similar is done for Pinot already in `PinotQueryGeneratorContext`

```
    public static final String TIME_BOUNDARY_FILTER_TEMPLATE = "__TIME_BOUNDARY_FILTER_TEMPLATE__";
    public static final String TABLE_NAME_SUFFIX_TEMPLATE = "__TABLE_NAME_SUFFIX_TEMPLATE__";
    public static final String DYNAMIC_FILTER_FUNCTION_TEMPLATE = "__DYNAMIC_FILTER_FUNCTION__";
```

Reply:
Okay, can you add a TODO to revisit this and/or add test cases that show current behavior if such a constant is used ?